### PR TITLE
deps(react-native): update to 0.73

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -79,7 +83,7 @@ post_install do |installer|
     use_native_modules![:reactNativePath],
     :mac_catalyst_enabled => false
   )
-  __apply_Xcode_12_5_M1_post_install_workaround(installer)
+#   __apply_Xcode_12_5_M1_post_install_workaround(installer)
```
Comment (saghul) [inline ios/Podfile:86]: Is this no longer necessary?

Comment (Calinteodor) [reply ios/Podfile:86]: https://github.com/facebook/react-native/commit/0ab8b40fd64ad86b4598293faa0b77e71fc9d349

File: package.json [hunk 2]
```diff
@@ -82,7 +82,7 @@
     "react-focus-on": "3.8.1",
     "react-i18next": "10.11.4",
     "react-linkify": "1.0.0-alpha",
-    "react-native": "0.72.14",
```
Comment (saghul) [inline package.json:85]: Can you update to 0.73.8

File: patches/react-native+0.73.0.patch [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
++}
++
+ - (void)stopTimers
+ {
+   if (_inBackground) {
+diff --git a/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.h b/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.h
+new file mode 100644
+index 0000000..e95b895
+--- /dev/null
++++ b/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.h
```
Comment (saghul) [inline patches/react-native+0.73.0.patch:60]: Drop this chunk.

File: patches/react-native+0.73.0.patch [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
++}
++#endif
++
++#pragma GCC diagnostic pop
++
+diff --git a/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.mm b/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.mm
+new file mode 100644
+index 0000000..a6c8eb5
+--- /dev/null
++++ b/node_modules/react-native/React/Fabric/RCTThirdPartyFabricComponentsProvider.mm
```
Comment (saghul) [inline patches/react-native+0.73.0.patch:99]: Drop this chunk.

File: patches/react-native+0.73.0.patch [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+diff --git a/node_modules/react-native/ReactCommon/react/renderer/components/rncore/States.h b/node_modules/react-native/ReactCommon/react/renderer/components/rncore/States.h
+new file mode 100644
+index 0000000..e69de29
+diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
+new file mode 100644
+index 0000000..361f5fb
+--- /dev/null
++++ b/node_modules/react-native/scripts/.packager.env
+@@ -0,0 +1 @@
++export RCT_METRO_PORT=8081
```
Comment (saghul) [inline patches/react-native+0.73.0.patch:168]: Drop this chunk.

File: ios/app/app.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1021,10 +1039,15 @@
 				ONLY_ACTIVE_ARCH = YES;
 				OTHER_CFLAGS = "$(inherited)";
 				OTHER_CPLUSPLUSFLAGS = "$(inherited)";
-				OTHER_LDFLAGS = "$(inherited)";
+				OTHER_LDFLAGS = (
+					"$(inherited)",
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply ios/app/app.xcodeproj/project.pbxproj:1043]: 15.2

File: package.json [hunk 7]
```diff
@@ -101,7 +101,7 @@
     "react-native-sound": "0.11.2",
     "react-native-splash-screen": "3.3.0",
     "react-native-svg": "13.13.0",
-    "react-native-svg-transformer": "1.1.0",
+    "react-native-svg-transformer": "1.2.0",
```
Comment (Calinteodor) [inline package.json:104]: This update fixed this issue 
https://github.com/kristerkari/react-native-svg-transformer/issues/317

Comment (saghul) [reply package.json:104]: Great!

File: package.json [hunk 8]
```diff
@@ -146,7 +146,6 @@
     "@types/react": "17.0.14",
     "@types/react-dom": "17.0.14",
     "@types/react-linkify": "1.0.1",
-    "@types/react-native": "0.69.22",
```
Comment (Calinteodor) [inline package.json:149]: This package has been deprecated.

Comment (saghul) [reply package.json:149]: Are the types bundled with the react-native package now?

Comment (Calinteodor) [reply package.json:149]: Updating to "@types/react-native": "0.73.0" logs a warning that react-native provides its own type definitions, so we don't need @types/react-native installed anymore.

File: ios/Podfile [hunk 9]
```diff
@@ -79,7 +83,7 @@ post_install do |installer|
     use_native_modules![:reactNativePath],
     :mac_catalyst_enabled => false
   )
-  __apply_Xcode_12_5_M1_post_install_workaround(installer)
```
Comment (saghul) [inline ios/Podfile:82]: If this is no longer needed, let's delete it!

File: patches/react-native+0.72.14.patch [hunk 10]
```diff
@@ -1,55 +0,0 @@
-diff --git a/node_modules/react-native/React/CoreModules/RCTTiming.mm b/node_modules/react-native/React/CoreModules/RCTTiming.mm
```
Comment (saghul) [inline patches/react-native+0.72.14.patch:1]: We need this patch on 0.73 too. It might require some adjustments.

Comment (Calinteodor) [reply patches/react-native+0.72.14.patch:1]: I though this was fixed after 0.73.4.

Comment (Calinteodor) [reply patches/react-native+0.72.14.patch]: @saghul I added back the patch file.

Comment (saghul) [reply patches/react-native+0.72.14.patch]: Last I checked it landed on 0.74

Comment (Calinteodor) [reply patches/react-native+0.72.14.patch]: That comes after 0.73.8. I'll try to update.

Comment (saghul) [reply patches/react-native+0.72.14.patch]: Don't! It's a bigger update.

Comment (saghul) [reply patches/react-native+0.72.14.patch]: Let's stick to 0.73 as planned, and once 0,.75 is out we can update to 0.74.latest.

**Review by saghul — APPROVED** [review]
🚀

**Pull request merged**